File: [프로그래머스] 코딩테스트 입문_안전지대.js
function solution(board) {
    let arr = [];
    for (let i = 0; i < board[0].length; i++){
        for (let j = 0; j < board[0].length; j++){
            if (board[i][j] === 1){
                arr.push([i,j]);
            }
        }
    }
    let answer = arr.length;
    for (let k of arr){
        for (let l of [-1, 0, 1]){
            for (let m of [-1, 0, 1]){
                if ((0 <= (k[0]+l) && (k[0]+l) < board[0].length) && (0 <= (k[1]+m) && (k[1]+m) < board[0].length)){
                    if (board[k[0]+l][k[1]+m] === 0){
                        board[k[0]+l][k[1]+m] = 1
                        answer++;
                    }
                }
            }
        }
    }
    return board.length ** 2 - answer;
}

File: [프로그래머스] 코딩테스트 입문_안전지대.py
def solution(board):
    newlist = [[i,j] for i in range(len(board)) for j in range(len(board[i])) if board[i][j] == 1]
    answer = len(newlist)
    for n in newlist:
        for i in [-1, 0, 1]:
            for j in [-1, 0, 1]:
                if 0 <= n[0]+i < len(board) and 0 <= n[1]+j < len(board):
                    if board[n[0]+i][n[1]+j] == 0:
                        board[n[0]+i][n[1]+j] = 1
                        answer += 1
    return len(board)**2 - answer

-----------------------------------

def solution(board):
    newlist = [[i,j] for i in range(len(board)) for j in range(len(board[i])) if board[i][j]==1]
    answer = len(newlist)
    x_arr = [-1, -1, -1, 0, 0, 1, 1, 1]
    y_arr = [-1, 0, 1, -1, 1, -1, 0, 1]
    for x, y in newlist:
        for i in range(8):
            new_x = x + x_arr[i]
            new_y = y + y_arr[i]
            if 0 <= new_x < len(board) and 0 <= new_y < len(board):
                if board[new_x][new_y] == 0:
                    board[new_x][new_y] = 1
                    answer += 1
    return len(board)**2 - answer